src/Microsoft.Health.Fhir.Shared.Core/Features/Operations/MemberMatch/MemberMatchService.cs

```diff
@@ -73,12 +75,20 @@ public async Task<ResourceElement> FindMatch(ResourceElement coverage, ResourceE
                 using IScoped<ISearchService> search = _searchServiceFactory();
                 results = await search.Value.SearchAsync(searchOptions, cancellationToken);
             }
-
-            // Generic catch for any exception happened during search.
+            catch (SqlException ex)
```

Comment:
Could we change this from a generic SQL exception to the new SqlQueryPlanException? That way we don't need to add the SQL library to Core. We are trying to keep database dependencies out of the non-database packages.

Reply:
OK, I'll go ahead and make that change.

Reply:
@brendankowitz What are your thoughts here? Matt looked into this and we can't get a specific exception at this point, the action process runs later. So the options are to have this be caught in the general exception catch, or take a dependency on SQL for the exception type. I don't like having it caught in the general exception as that means the user losses useful data.
I also know we have run into this issue before specifically with the SQL exception type.